extensions/amp-lightbox/1.0/lightbox.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {forwardRef} from '../../../src/preact/compat';
+import {setStyle} from '../../../src/style';
+import {
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from '../../../src/preact';
+
+const ANIMATION_PRESETS = {
+  'fade-in': [{opacity: 0}, {opacity: 1}],
+  'fly-in-top': [
+    {opacity: 0, transform: 'translate(0,-100%)'},
+    {opacity: 1, transform: 'translate: (0, 0)'},
+  ],
+  'fly-in-bottom': [
+    {opacity: 0, transform: 'translate(0, 100%)'},
+    {opacity: 1, transform: 'translate: (0, 0)'},
+  ],
+};
+
+/**
+ * @param {!LightboxDef.Props} props
+ * @param {{current: (!LightboxDef.LightboxApi|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function LightboxWithRef(
+  {
+    id,
+    layout = 'nodisplay',
+    animateIn = 'fade-in',
+    closeButtonAriaLabel,
+    // eslint-disable-next-line no-unused-vars
+    scrollable, // (TODO: discussion)
+    children,
+    onOpen,
+    openOnLoad,
+    ...rest
+  },
+  ref
+) {
+  const lightboxRef = useRef(null);
+  const [show, setShow] = useState(openOnLoad);
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      open: () => {
+        setShow(true);
+      },
+      close: () => {
+        setShow(false);
+      },
+    }),
+    [setShow]
+  );
+
+  useEffect(() => {
+    const element = lightboxRef.current;
+    if (show) {
```

Comment:
Technically it can be null if we decide to not display/render an element if the lightbox toggle is off (aka we do a `show && <ContainWrapper ...>`). I agree it's worth adding this check just in case, even if we don't go that route